src/libcryptobox/cryptobox.c

```diff
@@ -1586,16 +1635,23 @@ unsigned int rspamd_cryptobox_signature_bytes(enum rspamd_cryptobox_mode mode)
 #else
 #if OPENSSL_VERSION_MAJOR >= 3
 		if (ssl_keylen == 0) {
-			EVP_PKEY *pkey = EVP_PKEY_new();
-			EVP_PKEY_CTX *pctx = EVP_PKEY_CTX_new(pkey, ENGINE_F_ENGINE_NEW);
-			EVP_PKEY_CTX_set_ec_paramgen_curve_nid(pctx, CRYPTOBOX_CURVE_NID);
-			g_assert(pctx != NULL);
+			OSSL_LIB_CTX *libctx = OSSL_LIB_CTX_new();
```

Comment:
`#define CRYPTOBOX_CURVE_NID NID_X9_62_prime256v1` - and this curve should remain the same!